internal/health_checks/ports_check.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package health_checks
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"strconv"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+)
+
+type PortsCheck struct{}
+
+func (c PortsCheck) Name() string {
+	return "Ports Check"
+}
+
+// checkPortAvailable listens in the provided socket and local provided network (tcp4, tcp6, ...)
+// and handles the errors if the port is already being used by another process.
+func checkPortAvailable(host string, port string, network string) (bool, error) {
```

Comment:
nit: checkIfPortAvailable or checkPortAvailability

Reply:
Addressed in https://github.com/GoogleCloudPlatform/ops-agent/pull/1051/commits/c9d0ff992038c6c9e4a3dc234e8ac76d81bade12 .